app/Modules/Coupon/Coupon.php

```diff
@@ -251,6 +257,12 @@ public function appliedCouponInProduct()
         $couponDiscountAmount = $coupon->discount_amount;
         $couponDiscountType = $coupon->discount_type;
 
+        //check if applied coupon once for this product
+
+        if ($productPrice < $originalProduct->sale_price) {
+            wp_send_json_error(['message' => 'You have already avail this coupon code']);
```

Comment:
remove this condition. use `$originalProduct->sale_price` as `$productPrice` value. and calculate all discount based on this. so new new discount code can be applied.